File: tasks/implementations/java/tasks/basic_blockchain.java
import java.nio.file.*;
import java.util.*;

class BasicBlockchain {
    static final long MOD = 4_294_967_291L;
    static final long MULT = 16_777_619L;

    public static void run(String size, String fixturesRoot) throws Exception {
        List<String> lines = Files.readAllLines(Path.of(fixturesRoot, "generated", "blockchain", size + ".txt"));
        long prevHash = 2_166_136_261L;
        long total = 0L;
        for (int index = 0; index < lines.size(); index++) {
            String line = lines.get(index);
            if (line.isBlank()) continue;
            long hashValue = prevHash;
            for (String token : line.trim().split("\\s+")) {
                hashValue = (hashValue * MULT + Long.parseLong(token) + index + 1L) % MOD;
            }
            total = (total + hashValue) % MOD;
            prevHash = hashValue;
        }
        System.out.println(total);
    }
}

File: tasks/implementations/node/tasks/basic_blockchain.js
const fs = require('fs');
const MOD = 4294967291n;
const MULT = 16777619n;
module.exports = function(size, fixturesRoot) {
  const lines = fs.readFileSync(`${fixturesRoot}/generated/blockchain/${size}.txt`, 'utf8').trim().split(/\r?\n/);
  let prevHash = 2166136261n;
  let total = 0n;
  for (let index = 0; index < lines.length; index++) {
    let hashValue = prevHash;
    for (const token of lines[index].split(/\s+/)) {
      hashValue = (hashValue * MULT + BigInt(token) + BigInt(index + 1)) % MOD;
    }
    total = (total + hashValue) % MOD;
    prevHash = hashValue;
  }
  console.log(total.toString());
};
